synapse/storage/data_stores/main/events.py

```diff
@@ -1711,9 +1712,16 @@ def _purge_room_txn(self, txn, room_id):
 
         logger.info("[purge] done")
 
-    def purge_unreferenced_state_groups(self, room_id, state_groups_to_delete):
+    def purge_unreferenced_state_groups(
+        self, room_id: str, state_groups_to_delete: Set[int]
+    ) -> defer.Deferred:
         """Deletes no longer referenced state groups and de-deltas any state
         groups that reference them.
+
+        Args:
+            room_id: The room the state groups belong to (must all be in the
+                same room).
+            state_groups_to_delete: Set of all state groups to delete.
```

Comment:
we won't delete all of them. "Set of state groups to consider for deletion" or something.

Reply:
It does?

Reply:
uh, so it does